File: 프로그래머스/1/42840. 모의고사/모의고사.swift
import Foundation

func solution(_ answers:[Int]) -> [Int] {
    
    let p1 = [1, 2, 3, 4, 5]
    let p2 = [2, 1, 2, 3, 2, 4, 2, 5]
    let p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    var score = [0, 0, 0]
    
    for (i, answer) in answers.enumerated() {
        if answer == p1[i % p1.count] { score[0] += 1 }
        if answer == p2[i % p2.count] { score[1] += 1 }
        if answer == p3[i % p3.count] { score[2] += 1 }
    }
    
    let m = score.max()!
    
    return [1, 2, 3].filter { score[$0-1] == m }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
# 

def solution(answers):
    
    p1 = [1, 2, 3, 4, 5]
    p2 = [2, 1, 2, 3, 2, 4, 2, 5]
    p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    p1 = p1 * (int(10000 / len(p1)) + 1)
    p2 = p2 * (int(10000 / len(p2)) + 1)
    p3 = p3 * (int(10000 / len(p3)) + 1)
    
    s1, s2, s3 = 0, 0, 0
    
    for i in range(len(answers)):
        s1 += 1 if answers[i] == p1[i] else 0
        s2 += 1 if answers[i] == p2[i] else 0
        s3 += 1 if answers[i] == p3[i] else 0
    
    m = max(s1, s2, s3)
    answer = []
    if s1 == m:
        answer.append(1)
    if s2 == m:
        answer.append(2)
    if s3 == m:
        answer.append(3)
    
    return answer